service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/LocalAuthorityEmailContentProvider.java

```diff
@@ -35,4 +38,12 @@ public LocalAuthorityEmailContentProvider(LocalAuthorityNameLookupConfiguration
             .put("title", localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode))
             .build();
     }
+
+    public Map<String, Object> buildNoticeOfPlacementOrderUploadedNotification(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        return Map.of(
+            "respondentLastName", getFirstRespondentLastName(caseData.getRespondents1()),
+            "caseUrl", formatCaseUrl(uiBaseUrl, caseDetails.getId()));
+    }
```

Comment:
I do very similar in `PlacementApplicationContentProvider`. Could we clean it to re-use your more generic version or is this for the tech debt list?

Reply:
Assuming content can change, I think better to keep them apart. But yes, logic is identical 🙄 maybe tech debt, I hate the way emails and the content is organised